nalgebra-sparse/src/cs.rs

```diff
@@ -651,10 +639,9 @@ where
                     }
                     if prev > minor_idx {
                         if !sort {
-                            return Err(SparseFormatError::from_kind_and_msg(
-                                SparseFormatErrorKind::InvalidStructure,
-                                "Minor indices are not monotonically increasing within each lane.",
-                            ));
+                            panic!(
+                                "Minor indices are not monotonically increasing within each lane while sorting is not expected." 
+                            );
```

Comment:
Why was this turned into a panic?

Reply:
Sorry. I thought I replied to your [question](https://github.com/dimforge/nalgebra/pull/1015#discussion_r780163049) regarding testing this branch but can't find my comment.
I was going to add a test for this case and came to the problem that it is not reachable for tests at the moment as this function i being called by other methods in `csr.rs`/`csc.rs`. So decided to turn the error into panic as it looks like to be an _internal error_ so far. Do you think, we should still return an error here instead?

Reply:
In addition to turning this back into an error, I think for clarity of code, it would be helpful if the check here only marked the lane as nonmontonic and the actual further processing of that logic happens together with the nonmonotonic logic.

See also separate comment about needing to check for duplicates *after* sorting.

That is, something like this (pseudo)

```rust
let mut monotonic = true;

for minor_idx in minor_idx_in_lane {
    // check out of bounds
    // ..
    if prev >= minor_idx {
        monotonic = false;
    }
}

if !monotonic {
    if sort {
       sort indices and possibly values
    } else {
       return error "minor indices not strictly monotonic"   
    }
    check for duplicates
}
```

Reply:
I turned it back into error, but I still think that this case is not reachable for user with current implementation.